res/css/structures/_SearchBox.scss

```diff
@@ -15,7 +15,7 @@ limitations under the License.
 */
 
 .mx_SearchBox {
-    flex: 1 1 0;
+    flex: 1 1 0 !important;
```

Comment:
Instead of this, I think let's try the more "ambitious" version: leave this as-is, and remove `flex` from the `.mx_textinput` block in `_common.scss`.

I _believe_ the `.mx_textinput` currently only affects the room list filter, room history search, and member list search, and they all appear to look okay when testing this approach.

Reply:
@jryans I'm not totally clear on which line should be removed. It might be a better idea to close this PR and you can open one.

Removing each one of these lines individually each broke something:
https://github.com/matrix-org/matrix-react-sdk/blob/bb1d52919ad5c60f59bca6cef0b10c21f17480f6/res/css/_common.scss#L158
https://github.com/matrix-org/matrix-react-sdk/blob/bb1d52919ad5c60f59bca6cef0b10c21f17480f6/res/css/_common.scss#L173
https://github.com/matrix-org/matrix-react-sdk/blob/bb1d52919ad5c60f59bca6cef0b10c21f17480f6/res/css/_common.scss#L177

Reply:
Ah, I was thinking of the `flex: 0 0 auto` that you said the `!important` was required to override. What was broken when removing it?

Reply:
If you remove that line then the search box for searching community members becomes massive
<img width="1386" alt="screenshot" src="https://user-images.githubusercontent.com/5855073/100681668-de42bc00-3339-11eb-817a-4b5a8f39cec1.png">

Reply:
Ah okay, then let's proceed with that route, but also create a specific style for `mx_GroupMemberList_query` and `mx_GroupRoomList_query` with the `flex: 0 0 auto` they are relying on. In general, I think it's best for this `flex` value to be tuned for each component, since it depends on the containers it is inside. (It's a bit humorous, because there's already a specific style for these community inputs which was setting `flex: 1 1 0`, but the UX is actually relying on that being ignored and using the `mx_textinput` rule instead, which is very confusing!)

Reply:
Okay I think I've done that